pybaseball/playerid_lookup.py

```diff
@@ -47,77 +50,116 @@ def chadwick_register(save: bool = False) -> pd.DataFrame:
-            last = player.split(",")[0].strip()
-            first = None
-            if (len(player.split(",")) > 1):
-                first = player.split(",")[1].strip()
-            if (player_counter == 1):
-                results = playerid_lookup(last, first)
-            else:
-                results = results.append(playerid_lookup(last, first), ignore_index=True)
-            player_counter += 1
+def name_similarity(last: str, first: str, player_table: pd.DataFrame) -> pd.DataFrame:
+    """Calculates similarity of first and last name provided with all players in player_table
+
+    Args:
+        last (str): Provided last name
+        first (str): Provided first name
+        player_table (pd.DataFrame): Chadwick player table including names
+
+    Returns:
+        pd.DataFrame: 5 nearest matches from fuzzywuzzy.process
+    """
+    filled_df = player_table.fillna("")
+    chadwick_names = filled_df["name_first"] + " " + filled_df["name_last"]
+    fuzzy_matches = pd.DataFrame(process.extract(f"{first} {last}", chadwick_names, limit=5))
+    matched_names = fuzzy_matches[0].str.split(expand=True)
+    matched_names = matched_names.rename(columns={0: "name_first", 1: "name_last"})
+    return matched_names
+
+
+class player_search_client:
+    def __init__(self) -> None:
+        self.table = get_lookup_table()
+
+    def search(self, last: str = None, first: str = None, player_list: List[str] = None, fuzzy: bool = False) -> pd.DataFrame:
+        """Lookup playerIDs (MLB AM, bbref, retrosheet, FG) for a given player
+
+        Args:
+            last (str, optional): Player's last name. Defaults to None.
+            first (str, optional): Player's first name. Defaults to None.
+            player_list (list, optional): List of players to seach for (comma delimited). Defaults to None.
+            search (bool, optional): In case of typos, returns players with names close to input. Defaults to False.
+
+        Returns:
+            pd.DataFrame: DataFrame of playerIDs, name, years played
+        """
+
+        # force input strings to lowercase
+        if last:
+            last = last.lower()
+        if first:
+            first = first.lower()
+
+        # if player_list has a value, then the user is passing in a list of players
+        # the list of players may be comma delimited for last, first, or just last
+        if player_list:
+            player_counter = 1
+            for player in player_list:
+                last = player.split(",")[0].strip()
+                first = None
+                if (len(player.split(",")) > 1):
+                    first = player.split(",")[1].strip()
+                if (player_counter == 1):
+                    results = self.search(last, first)
+                else:
+                    results = results.append(self.search(last, first), ignore_index=True)
+                player_counter += 1
+            return results
+
+        if first is None:
```

Comment:
this if else block implies that supplying first name without last name is an invalid usage. should we make `last` a required argument?

Reply:
See above, I don't think so since list is possible